drivers: entropy: Several small fixes

ceolin opened the pull request:
- Fix return value in get_entropy_isr callback in some drivers
- Do not call `k_yield` from ISR

Commit:
entropy: gecko_se: Remove unnecessary callback

get_entropy_isr is not implemented, just remove it and let the
driver interface handle it.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
entropy: b91_trng: Fix callback return

get_entropy_isr() has to return the number of bytes copied or a negative
value for error. Since this driver is assuming that it will always
(????) get the number of requested bytes, change the function to return it
instead of 0.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
entropy: neorv32: Return ENODATA on error

Return -ENODATA in neorv32_trng_get_entropy_isr when
there is no data available. This is consistent with other
drivers.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

Commit:
entropy: fake_entropy: Fix return value in isr callback

get_entropy_isr() has to return the number of bytes copied or
a negative value in case of error.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

aescolar commented on drivers/entropy/entropy_sam.c:
```diff
@@ -58,14 +58,7 @@ static int entropy_sam_wait_ready(Trng * const trng, uint32_t flags)
-			 * do busy-wait for the random data to be ready.
-			 */
-			k_yield();
+			return -ENODATA;
```
The ATMEL maintainer may want to comment, but I think you mixed what is happening here.
The isr version seems to **never** call with `(flags & ENTROPY_BUSYWAIT) == 0` into here, and therefore this path will never be taken from ISR.

With your change to line 106 this `if` is never true in any case; and the code will busy wait instead of yielding if not `_ready` (when called in thread mode)

ceolin replied:
hmm, good catch. Actually the thing that needs to be changed here is just remove this k_yield() since the expected behavior is to block.

aescolar reviewed (dismissed):
Ok for the native driver changes, but the ATMEL changes do not seem correct.

Commit:
entropy: sam: Fix get_entropy() behavior

entropy_get_entropy blocks if required to generate the necessary
random data.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

nandojve commented on drivers/entropy/entropy_sam.c:
```diff
@@ -56,17 +56,6 @@ static int entropy_sam_wait_ready(Trng * const trng, uint32_t flags)
 			return -ETIMEDOUT;
 		}
-
-		if ((flags & ENTROPY_BUSYWAIT) == 0U) {
```
It seems flags is not necessary anymore so I was wondering if it is not possible to remove it.

ceolin replied:
I could be used in future, but that is just an internal function. Will remove it and if needed we re-introduce it later.

Commit:
entropy: sam: Remove unused parameter

Do not propagate unused parameter. ISR callback is already handling
the given flags, there is not need to propagate it through internal
calls.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

nandojve reviewed (approved):
Thank you @ceolin
